adapters/sn2core/sn2core.go

```diff
@@ -281,28 +281,52 @@ func AdaptCairo1Class(response *starknet.SierraDefinition, compiledClass json.Ra
 	}
 
 	if compiledClass != nil {
-		if err = json.Unmarshal(compiledClass, &class.Compiled); err != nil {
+		adaptedCompiledClass, err := AdaptCompiledClass(compiledClass)
+		if err != nil {
 			return nil, err
 		}
+		class.Compiled = *adaptedCompiledClass
 	}
+
 	return class, nil
 }
 
+func AdaptCompiledClass(compiledClass *starknet.CompiledClass) (*core.CompiledClass, error) {
+	compiled := new(core.CompiledClass)
```

Comment:
same here, unnecessarily allocated on heap. Caller immediately de-references it.

Reply:
same as above

Reply:
In this case there is actually -1 allocation if there is no `new`. Rewrote it